astropy/units/core.py

```diff
@@ -113,14 +114,25 @@ class _UnitRegistry(object):
     Manages a registry of the enabled units.
     """
     def __init__(self, init=[], equivalencies=[]):
-        if isinstance(init, _UnitRegistry):
-            equivalencies = init.equivalencies
-            init = init.all_units
-
         self._reset_units()
```

Comment:
These two lines can also be moved to the `else` clause.

Which means the two branches are completely separate, which in turn makes me wonder if it wouldn't be cleaner to implement a `copy` method and use that instead. But maybe not quite worth the effort (and can be done separately anyway).